base/printf.jl

```diff
@@ -468,7 +468,7 @@ function gen_e(flags::ASCIIString, width::Int, precision::Int, c::Char, inside_g
     if precision > 0
         if inside_g && !('#' in flags)
             push!(blk.args, :(endidx = $ndigits;
-                              while endidx > 1 && DIGITS[endidx] == '0'
+                              while endidx > 1 && DIGITS[endidx] == 0x30
```

Comment:
Please write this as `UInt8('0')`, that's much clearer. Or even better `Char(DIGITS[endidx]) == '0'`.